docs/porting/custom-target-porting.md

```diff
@@ -0,0 +1,259 @@
+
+You may be wondering why there are more directory levels than target configuration levels. This is because many targets use the `extra_labels_add` feature in the target configuration. The following keywords `STM32L4`, `STM32L475xG`, `STM32L475VG` resolve to `TARGET_STM32L4`, `TARGET_STM32L475xG`, `TARGET_STM32L475VG` respectively. With those labels applied, these directory names get included in the build for this target.
+
+1. Create a new directory called `TARGET_IMAGINARYBOARD` at the top level of your project.
+   
+   Your directory should look something like this:
+   
+   ```
+   custom_target.json
+   TARGET_IMAGINARYBOARD
+   mbed-os
+   .mbed
+   mbed_settings.py
+   mbed-os.lib
+   ```
+   
+1. Now place source files for the board.
+   
+   Inspect the files at `mbed-os\targets\TARGET_STM\TARGET_STM32L4\TARGET_STM32L475xG\TARGET_DISCO_L475VG_IOT01A`. You should find the following files or similar:
+   
+   `PeripheralNames.h`, `PeripheralPins.c`, `PinNames.h`, `system_clock.c`
+   
+   Copy the files into your new `TARGET_IMAGINARYBOARD` directory.
+   
+   The files provide these capabilities.
+      
+   - `PeripheralNames.h` describes the available peripherals and their base addresses.
+   - `PeripheralPins.c` describes the available pins and their association with peripherals.
+   - `PinNames.h` sets macros for pins that define their function.
+   - `system_clock.c` vendor specific file that initializes the system and sets up the clocks.
+      
+1. Modify the files.
+
+   `PinNames.h` is the most common file to be edited. For our example, the ImaginaryBoard is using I2C but connected to different supported signals. Change the I2C pin macro definitions from:
+   
+   ```
+   I2C_SCL     = D15,
+   I2C_SDA     = D14,
+   ```
+   
+   to
+   
+   ```
+   I2C_SCL     = PC_0,
+   I2C_SDA     = PC_1,
+   ```
+   
+   This edit is mainly for convenience if your application code uses standard I2C pin names `I2C_SDA` and `I2C_SCL`.
+   
+   You may also choose to add or remove peripherals, add or remove pins or change the clock frequency by editing `PeripheralNames.h`, `PeripheralPins.c` or `system_clock.c`. In our example, for simplicity, we will not edit these files.
+   
+1. (Optional) Add additional files.
+
+   If necessary, add additional source files for drivers or middleware you have implemented for the new board. In our example, we do not have any files to add.
+
+1. (Optional) Add a simple application source file for testing.
+
+    To confirm the software builds for the new target, add a file named `main.cpp` with the following contents:
+
+    ```  
+    #include "mbed.h"
+    
+    DigitalOut led1(LED1);
+
+    int main()
+    {
+        while (true) {
+            led1 = !led1;
+            wait_ms(500);
+        }
+    }
+    ``` 
+
+    This blinks an LED. Just make sure LED1 is defined for the board.
```

Comment:
"Just make sure LED1 is defined" is maybe an important point that should be made more prominent.